pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,101 @@
+package checks
+
+import (
+	"fmt"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// ProcessDiscovery is a ProcessDiscoveryCheck singleton. ProcessDiscoveryCheck should not be instantiated elsewhere.
+var ProcessDiscovery = &ProcessDiscoveryCheck{probe: procutil.NewProcessProbe()}
+
+// ProcessDiscoveryCheck is a check that gathers basic process metadata and sends it to the process discovery service.
+// It uses its own ProcessDiscovery payload, which is intended to be read by the process_discovery kafka topic.
+// The goal of this check is to collect information about possible integrations that may be enabled by the end user.
+type ProcessDiscoveryCheck struct {
+	probe      *procutil.Probe
+	info       *model.SystemInfo
+	initCalled bool
+}
+
+// Init initializes the ProcessDiscoveryCheck. It is a runtime error to call Run without first having called Init.
+func (d *ProcessDiscoveryCheck) Init(_ *config.AgentConfig, info *model.SystemInfo) {
+	d.info = info
+	d.initCalled = true
+}
+
+// Name returns the name of the ProcessDiscoveryCheck. In this case it is "process_discovery"
```

Comment:
nit: Let's remove this sentence as it has the potential to change in the future and the code will direct us to the actual value. 
```suggestion
// Name returns the name of the ProcessDiscoveryCheck.
```